### ISSUE TITLE
Boolean attribute values not of boolean type in WFS GetFeature GeoJSON response

### ISSUE THREAD

**Issue by arbakker** [MapServer/MapServer#6801]
## Expected behavior and actual behavior.

Boolean attribute values in WFS GetFeature GeoJSON response not correct when using `gml_*_type = Boolean` on the boolean attributes. I tested this for both GeoPackage and Postgis datasources, both datasources give incorrect results. 

Example datasource (where `bool_0` has value `0` for al features and `bool_1` value `1` for all features):

```sh
$ ogrinfo ahn3.gpkg ahn3_bladindex -so
...
FID Column = fid
Geometry Column = geom
puuid: String (0.0)
bladnr: String (24.0)
bool_0: Integer(Boolean) (0.0)
bool_1: Integer(Boolean) (0.0)
```

Example mapfile (see attached [zipfile](https://github.com/MapServer/MapServer/files/10443777/ahn3-boolean-issue.zip)) has two layers configured:

- ahn3_bladindex_gpkg (with a GeoPackage datasource)
- ahn3_bladindex_postgis (with a Postgis datasource)

For both layers the `gml_*_type` config option specified to indicate the boolean attribute type for the boolean attributes: 

```
# "gml_types"                             "auto"
"gml_bool_0_type"                     "Boolean" # has value 0 for all features
"gml_bool_1_type"                     "Boolean" # has value 1 for all features
```

Example [GeoJSON GetFeature](http://localhost:8080/mapserver?request=GetFeature&service=WFS&version=2.0.0&srsname=EPSG:28992&typenames=ahn3:ahn3_bladindex_gpkg&bbox=-234141.2978,153578.31569999998,546883.9822,770313.0357000001,EPSG:28992&count=1&outputFormat=application/json) response GPKG:

```json
{
  "type": "FeatureCollection",
  "name": "ahn3_bladindex_gpkg",
  ...
  "features": [
    {
      "type": "Feature",
      "id": "1",
      "properties": { "bladnr": "02ez1", "bool_0": 0, "bool_1": 1 },
      "geometry": {
       ...
      }
    }
  ]
}

```

Example [GeoJSON GetFeature](http://localhost:8080/mapserver?request=GetFeature&service=WFS&version=2.0.0&srsname=EPSG:28992&typenames=ahn3:ahn3_bladindex_postgis&bbox=-234141.2978,153578.31569999998,546883.9822,770313.0357000001,EPSG:28992&count=1&outputFormat=application/json) response PostGIS (note that the boolean values have value of `0` regardless of actual value in datasource):

```json
{
  "type": "FeatureCollection",
  "name": "ahn3_bladindex_postgis",
  ...
  "features": [
    {
      "type": "Feature",
      "id": "2",
      "properties": { "bladnr": "02ez2", "bool_0": 0, "bool_1": 0 },
      "geometry": {
       ...
      }
    }
  ]
}
```

The DescribeFeatureType request does correctly report that it is a boolean field (for both GeoPackage and PostGIS datasources):

```xml
<complexType name="ahn3_bladindex_postgisType">
    <complexContent>
      <extension base="gml:AbstractFeatureType">
        <sequence>
          <element name="geom" type="gml:GeometryPropertyType" minOccurs="0" maxOccurs="1"/>
          <element name="puuid" minOccurs="0" type="string"/>
          <element name="bladnr" minOccurs="0" type="string"/>
          <element name="bool_0" minOccurs="0" type="boolean"/>
          <element name="bool_1" minOccurs="0" type="boolean"/>
        </sequence>
      </extension>
    </complexContent>
  </complexType>
```

The xml GetFeature output does produce the correct [boolean values](http://www.datypic.com/sc/xsd/t-xsd_boolean.html), although they do differ between GeoPackage and PostGIS datasource:

[GeoPackage GetFeature](http://localhost:8080/mapserver?request=GetFeature&service=WFS&version=2.0.0&srsname=EPSG:28992&typenames=ahn3:ahn3_bladindex_gpkg&bbox=-234141.2978,153578.31569999998,546883.9822,770313.0357000001,EPSG:28992&count=1) response:

```xml
<ahn3:bladnr>02ez1</ahn3:bladnr>
<ahn3:bool_0>0</ahn3:bool_0>
<ahn3:bool_1>1</ahn3:bool_1>
```

[PostGIS GetFeature](http://localhost:8080/mapserver?request=GetFeature&service=WFS&version=2.0.0&srsname=EPSG:28992&typenames=ahn3:ahn3_bladindex_postgis&bbox=-234141.2978,153578.31569999998,546883.9822,770313.0357000001,EPSG:28992&count=1) response:

```xml
<ahn3:bladnr>02ez1</ahn3:bladnr>
<ahn3:bool_0>false</ahn3:bool_0>
<ahn3:bool_1>true</ahn3:bool_1>
```

###  Expected Behaviour

I would expect that the `has_data_*` fields would return a JSON boolean value (`true`, `false`):

```json
 {
      "type": "Feature",
      "id": "2",
      "properties": { "bladnr": "02ez2", "bool_0": false, "bool_1": true },
      "geometry": {
       ...
      }
```

## Steps to reproduce the problem.

- unzip attached zip archive
- start `docker-compose.yaml` file from unzipped zip archive
- do GetFeature requests with outputFormat GeoJSON:
    - GeoPackage: http://localhost:8080/mapserver?request=GetFeature&service=WFS&version=2.0.0&srsname=EPSG:28992&typenames=ahn3:ahn3_bladindex_gpkg&bbox=-234141.2978,153578.31569999998,546883.9822,770313.0357000001,EPSG:28992&count=1&outputFormat=application/json
    - PostGIS: http://localhost:8080/mapserver?request=GetFeature&service=WFS&version=2.0.0&srsname=EPSG:28992&typenames=ahn3:ahn3_bladindex_postgis&bbox=-234141.2978,153578.31569999998,546883.9822,770313.0357000001,EPSG:28992&count=1&outputFormat=application/json

## Attach simple test case (drag/drop it here)

[ahn3-boolean-issue.zip](https://github.com/MapServer/MapServer/files/10443777/ahn3-boolean-issue.zip)

## Operating system

See [camptocamp/docker-mapserver](https://github.com/camptocamp/docker-mapserver/blob/d4ec0ae0227da1ed83f3d57bafd9291c5b90724f/Dockerfile#L69)

## MapServer version and installation method

- [camptocamp/mapserver:8.0](https://github.com/camptocamp/docker-mapserver) docker image

**Comment by geographika** [comment]
@arbakker - #6793 got merged into main yesterday, and fixes Boolean fields when using `gml_types` `auto` for PostGIS. The responses and values for GetFeature all seem correct in the tests. Maybe this is a different issue when setting field types manually, but maybe give `gml_types` `auto` a try with the latests MapServer main branch.

**Comment by arbakker** [comment]
I just checked whether #6793 fixes the issue but it does not, since that issue only addresses the Boolean values in the GML output and not the GeoJSON output. 

So here is what I did:

```
git clone git@github.com:camptocamp/docker-mapserver.git
cd docker-mapserver
export DOCKER_TAG=latest-local; make build && cd ..
docker-compose up # also update the docker tag to latest-local in the docker-compose file
```

I made the following overview of the two config option with the four output-datasource combinations; YES/NO indicates whether the boolean value is outputted correct:

|                            |    gml-output postgis ds | gml-output geopackage ds | geojson-output postgis ds | geojson-output-geopackage ds |
| -------------------------- | --------------------- | ---------------------- | ---------------------- | ------------------------- |
| latest main `gml_types auto` |                   YES |                     NO |                     NO |                        NO |
| latest main `gml_*_type`    |                   YES |                    YES |                     NO |                        NO |

So seems that #6793 only fixes the GML output for PostGIS datasource when using `gml_types auto`.  It does not fix it for a GeoPackage datasource. Nor does it address the GeoJSON output (for both GeoPackage and PostGIS datasource).

**Comment by geographika** [comment]
@arbakker - would you be able to test the `ogr2ogr` output from your datasources to GeoPackage and GeoJSON, and that the field type outputs are correct in those?

**Comment by arbakker** [comment]
Yes, executed from the extracted zip archive with the docker-compose running:

GPKG datasource:

```sh
$ ogrinfo ahn3.gpkg ahn3_bladindex -so | grep bool_
bool_0: Integer(Boolean) (0.0)
bool_1: Integer(Boolean) (0.0)

$ ogr2ogr -f GeoJSON /vsistdout/ ahn3.gpkg | ogrinfo /vsistdin/ ahn3_bladindex -so | grep bool_
bool_0: Integer(Boolean) (0.0)
bool_1: Integer(Boolean) (0.0)
```
PostGIS datasource (which has been loaded tot the db with `ogr2ogr -f PostgreSQL PG:"dbname='gis' host='db' port='5432' user='postgres' password='postgres'" /data/ahn3.gpkg`, see `db-init-script.sh`):

```sh
$ PG_CONN=PG:"host=127.0.0.1 user=postgres password=postgres dbname=gis port=5432"
$ ogrinfo $PG_CONN ahn3_bladindex -so | grep bool_
bool_0: Integer(Boolean) (1.0)
bool_1: Integer(Boolean) (1.0)

$ gpkg_out=$(mktemp --suffix=.gpkg -u)
$ ogr2ogr -f GPKG $gpkg_out $PG_CONN && ogrinfo $gpkg_out ahn3_bladindex -so | grep bool_
bool_0: Integer(Boolean) (1.0)
bool_1: Integer(Boolean) (1.0)

ogr2ogr -f GeoJSON /vsistdout/ $PG_CONN | ogrinfo /vsistdin/ ahn3_bladindex -so | grep bool_
bool_0: Integer(Boolean) (1.0)
bool_1: Integer(Boolean) (1.0)
```

**Comment by geographika** [comment]
Fixed with #6803
